### ISSUE TITLE
Presearch - recover from networking errors during init

### ISSUE THREAD

**Issue by scottyeager** [threefoldtech/tfgrid-sdk-ts#130]
Recently I launched a Presearch deployment (on mainnet node 3995) and later noticed it hadn't appeared online on the Presearch dashboard. `zinit` reported `prerestorer: "Error(Exited(Pid(407), 1))"` and I found the following error in the logs:
```
[+] prerestorer: Unable to find image 'presearch/node:latest' locally
[+] dockerd: time="2022-07-29T18:07:03.081832466Z" level=debug msg="Calling GET /v1.41/info"
[+] dockerd: time="2022-07-29T18:07:03.131285453Z" level=debug msg="Calling POST /v1.41/images/create?fromImage=presearch%2Fnode&tag=latest"
[+] dockerd: time="2022-07-29T18:07:03.419097268Z" level=debug msg="Trying to pull presearch/node from https://registry-1.docker.io v2"
[+] dockerd: time="2022-07-29T18:07:13.525396835Z" level=warning msg="Error getting v2 registry: Get https://registry-1.docker.io/v2/: net/http: TLS handshake timeout"
[+] dockerd: time="2022-07-29T18:07:13.720370630Z" level=info msg="Attempting next endpoint for pull after error: Get https://registry-1.docker.io/v2/: net/http: TLS handshake timeout"
[+] dockerd: time="2022-07-29T18:07:13.919189138Z" level=error msg="Handler for POST /v1.41/images/create returned error: Get https://registry-1.docker.io/v2/: net/http: TLS handshake timeout"
[+] prerestorer: docker: Error response from daemon: Get https://registry-1.docker.io/v2/: net/http: TLS handshake timeout.
```

A simple `zinit start prerestorer` was enough to get everything running again. I guess there was some temporary network failure during the initial start up. Using `oneshot` with no retry for both the restorer and updater service creates fragility. Ideally, these should retry via some mechanism until they succeed.

**Comment by Omarabdul3ziz** [comment]
1. To fix the new issue (failure on registering the layer) we needed to add a docker disk at `/var/lib/docker` with just a minimal 10GB size. this raises the cost for preaserch deployment from `1.8` to `1.98` USD per month.
2. To fix the original issue (recovering from network issues) we added a new zinit service that will pull the image first before the container is run. the `puller` service includes a test command that will make sure that the image is downloaded and registered before starting the container. 
  Actually, I couldn't reproduce this issue in the first place but hopefully this fix will be enough to recover if issues happen during pulling the image. 


Note:
needed to update the flist version to be `v2.3` new flist `tf-official-apps/presearch-v2.3.flist` to not break the old one. so this will not affect the main net directly but will take the releasing process until `3.11` reach main net

**Comment by scottyeager** [comment]
Thanks @Omarabdul3ziz. The issue when pulling the container file is not easy to reproduce, since it happens on nodes with slower or less stable network connections.

As for the `failed to register layer` error, this can also be resolved just by user a more recent version of Docker than what comes in the Ubuntu repos. I typically use the script at `get.docker.com` to easily get an up to date version from Docker's repository. Currently that's `24.0.5` and it works well without needing to attach a disk.

One final note that's unrelated to this issue but I thought of since you mentioned the deployment cost—the Presearch node does not need 1gb of RAM. It runs fine with 512mb and the cost is less.

**Comment by mohamedamer453** [comment]
Verified on devnet.

A `10gb` disk was added at `/var/lib/docker`

![image](https://github.com/threefoldtech/tfgrid-sdk-ts/assets/101194226/a5f802d0-5f6f-42cc-ae3f-0843d247b8c2)

i've also tested multiple instance and all of them started successfully. 


![image](https://github.com/threefoldtech/tfgrid-sdk-ts/assets/101194226/d29a26f8-52b1-42b9-9f54-a57730743db9)

![image](https://github.com/threefoldtech/tfgrid-sdk-ts/assets/101194226/685ddf0a-d584-44ff-bcb2-0f79ffdf647e)

![image](https://github.com/threefoldtech/tfgrid-sdk-ts/assets/101194226/28918412-2049-48e2-9061-5738e6ba725e)

![image](https://github.com/threefoldtech/tfgrid-sdk-ts/assets/101194226/4faee399-54d4-4d9c-b446-395f0f62e419)

and both network options (planetary/public ipv4) are working without any issues.

I'll also keep these instances up for a while to verify that everything is working fine and will update/reopen this issue if there're any issues.
